python/tvm/relay/op/nn/nn.py

```diff
@@ -1471,7 +1471,46 @@ def bias_add(data, bias, axis=1):
     return _make.bias_add(data, bias, axis)
 
 
-def dense(data, weight, units=None, out_dtype=""):
+def matmul(data, weight, units=None, out_dtype="", data_transposed=False, weight_transposed=False):
+    """Dense operator.
+    Applies a linear transformation. The X & W can be transposed.
+
+    .. math::
+
+    `Y = X * W`
+
+    Parameters
+    ----------
+    data : tvm.relay.Expr
+        The input data to the operator,
+        of shape `(d_1, d_2, ..., d_n, units_in)` or `(d_1, d_2, ..., units_in, d_n)`.
+
+    weight : tvm.relay.Expr
+        The weight expressions, 2-D matrix,
+        of shape `(units_in, units)` or `(units, units_in)`.
+
+    units : int, optional
+        Number of hidden units of the dense transformation.
+
+    out_dtype : str, optional
+        Specifies the output data type for mixed precision dense,
+        of shape `(d_1, d_2, ..., d_n, units)`.
+
+    data_transposed : bool, optional
+        Whether the data tensor is in transposed format.
+
+    weight_transposed : bool, optional
+        Whether the weight tensor is in transposed format.
+
+    Returns
+    -------
+    result : tvm.relay.Expr
+        The computed result.
+    """
+    return _make.matmul(data, weight, units, out_dtype, data_transposed, weight_transposed)
+
+
+def dense(data, weight, units=None, out_dtype="", data_transposed=False, weight_transposed=True):
```

Comment:
It's weird to have transpose arguments in dense. This op should enforce transposed weight. Ideally, the implementation of dense in Relay level should be:

```
return _make.matmul(data, weight, units, out_dtype, False, True)
```

Reply:
TVM has some API calling `relay_op(xxx, **attrs)`, python function calls cannot work with the missing of the two parameters.